src/player.cpp

```diff
@@ -12124,6 +12072,63 @@ bool player::armor_absorb( damage_unit& du, item& armor )
     return armor.damage >= 5;
 }
 
+void player::bionic_armor_absorb( body_part bp, damage_unit &du ) const
+{
+    // We only check the passive bionics
+    if( has_bionic( "bio_carbon" ) ) {
+        if( du.type == DT_BASH ) {
+            du.amount -= 2;
+        } else if( du.type == DT_CUT ) {
+            du.amount -= 4;
+        } else if( du.type == DT_STAB ) {
+            du.amount -= 3.2;
+        }
+    }
+    //all the other bionic armors reduce bash/cut/stab by 3/3/2.4
+    // Map body parts to a set of bionics that protect it
+    // @todo: JSONize passive bionic armor instead of hardcoding it
+    static const std::map< body_part, std::set< std::string >  > armor_bionics = {
```

Comment:
Now that each set has only 1 element, the map can be changed to just `body_part->string`.